core/unit_test/tstAoSoA.hpp

```diff
@@ -474,6 +474,94 @@ void testAccess()
     checkDataMembers( aosoa, fval, dval, ival, dim_1, dim_2, dim_3 );
 }
 
+//---------------------------------------------------------------------------//
+// Manually defined SoA.
+template<int VLEN,int D1, int D2, int D3>
+struct MySoA
+{
+    float m0[D1][D2][D3][VLEN];
+    int m1[VLEN];
+    double m2[D1][VLEN];
+    double m3[D1][D2][VLEN];
+};
+
+// Test an unmanaged AoSoA.
+void testUnmanaged()
+{
+    // Manually set the inner array size.
+    const int vector_length = 16;
+
+    // Data dimensions.
+    const int dim_1 = 3;
+    const int dim_2 = 2;
+    const int dim_3 = 4;
+
+    // Declare data types that are equivalent to the user defined struct.
+    using DataTypes =
+        Cabana::MemberTypes<float[dim_1][dim_2][dim_3],
+                            int,
+                            double[dim_1],
+                            double[dim_1][dim_2]
+                            >;
+
+    // Allocate an AoSoA manually.
+    using soa_type = MySoA<vector_length,dim_1,dim_2,dim_3>;
+    int num_soa = 3;
+    int size = 35;
+    Kokkos::View<soa_type*,TEST_MEMSPACE> user_data( "user_aosoa", 3 );
```

Comment:
Would this be a better example if it came from totally outside of Kokkos (a user made array)?